pkg/ccip/provider/provider.go

```diff
@@ -20,26 +24,40 @@ var _ commontypes.CCIPProvider = &Provider{}
 const CCIPProviderName = "TONCCIPProvider"
 
 type Provider struct {
-	lggr logger.Logger
-	ca   ccipocr3.ChainAccessor
-	ct   ocr3types.ContractTransmitter[[]byte]
+	lggr  logger.Logger
+	ca    ccipocr3.ChainAccessor
+	ct    ocr3types.ContractTransmitter[[]byte]
+	codec ccipocr3.Codec
 
 	wg sync.WaitGroup
 	services.StateMachine
 }
 
-func NewCCIPProvider(lggr logger.Logger, txm txm.TxManager) (*Provider, error) {
+func NewCCIPProvider(lggr logger.Logger, chainSelector ccipocr3.ChainSelector, client ton.APIClientWrapped, txm txm.TxManager, logPoller logpoller.LogPoller) (*Provider, error) {
 	ct, err := ocr.NewCCIPTransmitter(txm, lggr)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a CCIP ContractTransmitter %w", err)
+		return nil, fmt.Errorf("failed to create a CCIP ContractTransmitter: %w", err)
 	}
 
-	cp := &Provider{
-		lggr: logger.Named(lggr, CCIPProviderName),
-		ct:   ct,
+	codec := ccipocr3.Codec{
+		ChainSpecificAddressCodec: &codec.AddressCodec{},
+		CommitPluginCodec:         &codec.CommitPluginCodecV1{},
+		ExecutePluginCodec:        &codec.ExecutePluginCodecV1{},
```

Comment:
There's an issue here, the executePluginCodec needs [extraDataCodec](https://github.com/smartcontractkit/chainlink/blob/b24c2e635e7dab7aefa1f4857c8c0588298d85a9/core/capabilities/ccip/ccipsolana/pluginconfig.go#L17) which is a map that gets initialized during runtime. We have to find a workaround here.